saleor/graphql/discount/mutations/voucher/voucher_create.py

```diff
@@ -98,39 +116,151 @@ class Meta:
 
     @classmethod
     def clean_input(cls, info: ResolveInfo, instance, data, **kwargs):
-        code = data.get("code", None)
-        if code == "":
-            data["code"] = generate_promo_code()
-        elif not is_available_promo_code(code):
-            raise ValidationError(
-                {
-                    "code": ValidationError(
-                        "Promo code already exists.",
-                        code=DiscountErrorCode.ALREADY_EXISTS.value,
-                    )
-                }
+        if data.code != "":
+            validate_one_of_args_is_in_mutation(
+                "code",
+                data.code,
+                "codes",
+                data.codes,
+                use_camel_case=True,
             )
+
+        cls.clean_codes(data)
         cleaned_input = super().clean_input(info, instance, data, **kwargs)
 
         return cleaned_input
 
     @classmethod
-    def success_response(cls, instance):
-        instance = ChannelContext(node=instance, channel_slug=None)
-        return super().success_response(instance)
+    def clean_codes(cls, data):
+        if data.code is not None:
+            if data.code == "":
```

Comment:
What if `code` is `" "`?
Maybe `if not data.code:` instead?

Reply:
Ok, and I added `strip()` for codes